src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java

```diff
@@ -443,17 +441,9 @@ public void checkCoupling() {
          * @return true if we should count this class.
          */
         private boolean isSignificant(String candidateClassName) {
-            boolean result = !excludedClasses.contains(candidateClassName)
-                && !isFromExcludedPackage(candidateClassName);
-            if (result) {
-                for (Pattern pattern : excludeClassesRegexps) {
-                    if (pattern.matcher(candidateClassName).matches()) {
-                        result = false;
-                        break;
-                    }
-                }
-            }
-            return result;
+            return !excludedClasses.contains(candidateClassName)
+                && !isFromExcludedPackage(candidateClassName)
+                && !isExcludedClassRegexp(candidateClassName);
```

Comment:
This basically amounts to a no change. The if statement was surviving, removing it meant we always went through the patterns even if we already knew it wasn't a match. Moving it to another method and chaining it to the &s allowed it the code to basically stay as it is.